polaris-terraform/ui-terraform/nginx.js

```diff
@@ -3,9 +3,10 @@ async function polarisAuthRedirect(r) {
   const cookie = encodeURIComponent(r.headersIn.Cookie ?? "")
   const referer = encodeURIComponent(r.headersIn.Referer ?? "")
   const polarisUiUrl = encodeURIComponent(r.args["polaris-ui-url"] ?? "")
+  const q = encodeURIComponent(r.args["q"] ?? "")
```

Comment:
shouldn't need to encode this, should already be encoded, so you might end up double encoding it
```suggestion
  const q = r.args["q"] ?? ""
```